src/main/java/spoon/support/compiler/jdt/JDTBasedSpoonCompiler.java

```diff
@@ -368,11 +368,16 @@ protected boolean buildUnitsAndModel(JDTBuilder jdtBuilder, SpoonFolder sourcesF
 	private static final CompilationUnitDeclaration[] EMPTY_RESULT = new CompilationUnitDeclaration[0];
 
 	protected CompilationUnitDeclaration[] buildUnits(JDTBuilder jdtBuilder, SpoonFolder sourcesFolder, String[] classpath, String debugMessagePrefix, boolean buildOnlyOutdatedFiles) {
-		List<SpoonFile> sourceFiles = Collections.unmodifiableList(sourcesFolder.getAllJavaFiles());
+		List<SpoonFile> sourceFiles = sourcesFolder.getAllJavaFiles();
 		if (sourceFiles.isEmpty()) {
 			return EMPTY_RESULT;
 		}
 
+		if (buildOnlyOutdatedFiles && outputDirectory.exists()) {
+			@SuppressWarnings("unchecked") Collection<File> outputFiles = FileUtils.listFiles(outputDirectory, new String[] { "java" }, true);
+			keepOutdatedFiles(sourceFiles, outputFiles);
```

Comment:
`keepOutdatedFiles` should not change the list of files but instead return the modified list. Its impact would be more clear.